### PR TITLE
Add metadata column to the migrations table

### PR DESCRIPTION
**Pull request by benjeffery**
Add migration metadata, note that like edges this is backward compatible as the columns are optional in the kastore.

### CODE REVIEW

**Comment by benjeffery** [comment]
Looks like I broke the docs, will fix later as AFK.

**Comment by benjeffery** [comment]
@jeromekelleher ready for a once over.

File: c/tskit/core.h [hunk 1]
```diff
@@ -96,7 +96,7 @@ to the API or ABI are introduced, i.e., internal refactors of bugfixes.
 #define TSK_FILE_FORMAT_NAME          "tskit.trees"
 #define TSK_FILE_FORMAT_NAME_LENGTH   11
 #define TSK_FILE_FORMAT_VERSION_MAJOR 12
-#define TSK_FILE_FORMAT_VERSION_MINOR 1
+#define TSK_FILE_FORMAT_VERSION_MINOR 2
```
Comment (jeromekelleher) [inline c/tskit/core.h]: Probably not worth bothering with the minor version bump here? We've not released it so I think it's reasonable to merge together all the compatible changes between releases into one minor version bump. Doesn't seem much point in having lots of transient file versions.

File: python/tskit/tables.py [hunk 2]
```diff
@@ -849,22 +880,50 @@ def set_columns(
         :type dest: numpy.ndarray, dtype=np.int32
         :param time: The time of each migration.
         :type time: numpy.ndarray, dtype=np.int64
+        :param metadata: The flattened metadata array. Must be specified along
+            with ``metadata_offset``. If not specified or None, an empty metadata
+            value is stored for each node.
```
Comment (jeromekelleher) [inline python/tskit/tables.py]: s/node/migration

File: python/tskit/tables.py [hunk 3]
```diff
@@ -878,9 +937,24 @@ def append_columns(self, left, right, node, source, dest, time):
...
```
Comment (jeromekelleher) [inline python/tskit/tables.py]: ditto

File: c/tskit/core.h (hunk 1 above)
Comment (benjeffery) [reply c/tskit/core.h]: Cool, as long one PR that needs to in a release increments should be good. I wasn't sure how many users we had who are on the bleeding edge.

**Comment by benjeffery** [comment]
@jeromekelleher Ok, all fixed.

File: c/tskit/core.h (hunk 1 above)
Comment (jeromekelleher) [reply c/tskit/core.h]: Trying to discourage bleeding edge users, but this is harmless anyway since the changes are backward/forward compatible.

**Comment by jeromekelleher** [comment]
Merged, thanks. OK, that's the boring stuff out of the way!

**Pull request merged**